Lib/ufo2ft/instructionCompiler.py

```diff
@@ -0,0 +1,280 @@
+
+        ttglyph.program = ttLib.tables.ttProgram.Program()
+        ttglyph.program.fromAssembly(asm)
+
+    def _set_composite_flags(self, glyph: Glyph, ttglyph: TTGlyph) -> None:
+        # Set component flags
+
+        if len(ttglyph.components) != len(glyph.components):
+            # May happen if nested components have been flattened by a filter
+            logger.error(
+                "Number of components differ between UFO and TTF "
+                f"in glyph '{glyph.name}' ({len(glyph.components)} vs. "
+                f"{len(ttglyph.components)}, not setting component flags from"
+                "UFO. They may still be set heuristically."
+            )
+            return
+
+        # We need to decide when to set the flags.
+        # Let's assume if any lib key is not there, or the component
+        # doesn't have an identifier, we should leave the flags alone.
+
+        # Keep track of which component has the USE_MY_METRICS flag
+        use_my_metrics_comp = None
+
+        for i, c in enumerate(ttglyph.components):
+
+            # Set OVERLAP_COMPOUND on the first component only
+            if i == 0 and TRUETYPE_OVERLAP_KEY in glyph.lib:
+                if glyph.lib.get(TRUETYPE_OVERLAP_KEY, False):
+                    c.flags |= OVERLAP_COMPOUND
+                else:
+                    c.flags &= ~OVERLAP_COMPOUND
+
+            # Check if we have information about the current component in the glyph lib
+            ufo_component_id = glyph.components[i].identifier
+            if ufo_component_id is None:
+                # No information about component flags is stored in the UFO.
+                # We don’t modify the flags. Two flags have already been set elsewhere:
+                # - ROUND_XY_TO_GRID is set in TTGlyphPointPen.glyph() called from
+                #                    OutlineTTFCompiler.compileGlyphs()
+                # - USE_MY_METRICS   is set in OutlineTTFCompiler.setupTable_glyf()
+                continue
+
+            if (
+                OBJECT_LIBS_KEY in glyph.lib
+                and ufo_component_id in glyph.lib[OBJECT_LIBS_KEY]
+                and (
+                    TRUETYPE_ROUND_KEY in glyph.lib[OBJECT_LIBS_KEY][ufo_component_id]
+                    or TRUETYPE_METRICS_KEY
+                    in glyph.lib[OBJECT_LIBS_KEY][ufo_component_id]
+                )
+            ):
+                component_lib = glyph.lib[OBJECT_LIBS_KEY][ufo_component_id]
+
+                # ROUND_XY_TO_GRID
+
+                # https://github.com/googlefonts/ufo2ft/pull/425 recommends
+                # to always set the ROUND_XY_TO_GRID flag, so we only
+                # unset it if explicitly done so in the lib
+                if not component_lib.get(TRUETYPE_ROUND_KEY, True):
+                    c.flags &= ~ROUND_XY_TO_GRID
+
+                # USE_MY_METRICS
+
+                if self.autoUseMyMetrics:
+                    # Leave the existing USE_MY_METRICS flag alone
```

Comment:
@jenskutilek why do we even need this `self.autoUseMyMetrics` option here? If the UFO glyph component has the useMyMetrics lib key set, it should just be used no matter whether "auto" mode would set it or not... Since a concrete user override is always more specific and should take priority over any automatic euristics (rather than the other way around). I suggest we get rid of that option altogether